dali/pipeline/operator/builtin/input_operator.h

```diff
@@ -35,6 +37,68 @@ struct CudaEventWrapper : CUDAEvent {
   CudaEventWrapper() : CUDAEvent(CUDAEvent::Create()) {}
 };
 
+
+/**
+ * Custom structure to hold the input data inside the CachingList.
+ * Apart from holding a pointer to the data, it can also contain other data associated with the
+ * input data, e.g. name, flavour, colour, etc.
+ * @tparam Backend
+ */
+template<typename Backend>
+struct named_pointer_to_tensor_list_t {
+  using element_type = TensorList<Backend>;
+  std::unique_ptr<element_type> data = nullptr;
+  std::optional<std::string> data_id = std::nullopt;
+
+
+  named_pointer_to_tensor_list_t(std::unique_ptr<element_type> data) :  // NOLINT
+          data(std::move(data)) {}
+
+  ~named_pointer_to_tensor_list_t() = default;
+
+  named_pointer_to_tensor_list_t(const named_pointer_to_tensor_list_t &) = delete;
+
+  named_pointer_to_tensor_list_t &operator=(const named_pointer_to_tensor_list_t &) = delete;
+
+
+  named_pointer_to_tensor_list_t &operator=(named_pointer_to_tensor_list_t &&other) noexcept {
+    data = std::move(other.data);
+    data_id = std::move(other.data_id);
+    other.data = nullptr;
+    other.data_id = std::nullopt;
+  }
+
+
+  named_pointer_to_tensor_list_t(named_pointer_to_tensor_list_t &&other) noexcept {
+    *this = std::move(other);
+  }
+
+
+  const auto &operator*() const noexcept {
+    return *data;
+  }
+
+
+  auto &operator*() noexcept {
+    return *data;
+  }
+
+
+  const auto &operator->() const noexcept {
+    return data;
+  }
+
+
+  auto &operator->() noexcept {
+    return data;
+  }
```

Comment:
This is not in line with how smart pointers work - the operators `*` and `->` are always const-qualified and the constness of the returned object pointer relies solely on the type of the object.
```C++
const unique_ptr<int> u = ...;
int *p = u.get();  // OK
```
Therefore, if you want to mimic unique_ptr, the code should be:
```suggestion
  auto &operator*() const noexcept {
    return *data;
  }


  auto &operator->() const noexcept {
    return data;
  }
```

Reply:
Done

Reply:
It's back?

Reply:
It is now. Sorry for that.